search-books/src/test/java/nix/education/functionals/ElasticsearchServiceTest.java

```diff
@@ -0,0 +1,235 @@
+
+            JsonNode responseBody =
+                MAPPER.readTree(EntityUtils.toString(response.getEntity(), "UTF-8"));
+            JsonNode expectedBody = readFileContent("all-books.json");
+
+            assertEquals(HttpStatus.SC_OK, response.getStatusLine().getStatusCode());
+            assertEquals(expectedBody.size(), responseBody.size());
+            assertEquals(expectedBody, responseBody);
+        }
+    }
+
+    @Nested
+    @DisplayName("\"addBook\" method should ")
+    public class AddBookTest {
+
+        @Test
+        @DisplayName("show bad request error")
+        public void shouldShowBadRequestErrorWhenAddIncorrectNewBook() throws Exception {
+            HttpPost request = new HttpPost("http://localhost:8080/add");
+            HttpResponse response = httpClient.execute(request);
+
+            assertEquals(HttpStatus.SC_BAD_REQUEST,
+                response.getStatusLine().getStatusCode());
+        }
+
+        @Test
+        @DisplayName("add new book")
+        public void shouldAddNewBook() throws Exception {
+            HttpPost request = new HttpPost("http://localhost:8080/add");
+            request.addHeader("Content-Type", "application/json");
+            String pathFile = "src/test/resources/add-book-request.json";
+            String requestEntity = new String(Files.readAllBytes(Paths.get(pathFile)));
+            StringEntity params = new StringEntity(requestEntity);
+            request.setEntity(params);
+            HttpResponse response = httpClient.execute(request);
+
+            JsonNode responseBody =
+                MAPPER.readTree(EntityUtils.toString(response.getEntity(), "UTF-8"));
+            JsonNode expectedBody = readFileContent("add-book-response.json");
+
+            HttpDelete requestDelete =
+                new HttpDelete("http://localhost:8080/delete/" + responseBody.get("id"));
+            httpClient.execute(requestDelete);
+
+            assertEquals(expectedBody.get(AUTHOR), responseBody.get(AUTHOR));
+            assertEquals(expectedBody.get(TITLE), responseBody.get(TITLE));
+            assertEquals(expectedBody.get(DESCRIPTION), responseBody.get(DESCRIPTION));
+            assertEquals(expectedBody.get(TAGS), responseBody.get(TAGS));
+            assertEquals(expectedBody.get(PUBLICATION_DATE), responseBody.get(PUBLICATION_DATE));
+            assertEquals(HttpStatus.SC_CREATED, response.getStatusLine().getStatusCode());
+        }
+    }
+
+    @Nested
+    @DisplayName("\"deleteBook\" method should ")
+    public class DeleteBookTest {
+
+        @Test
+        @DisplayName("delete book by id")
+        public void shouldDeleteBookById() throws Exception {
+            HttpPost request = new HttpPost("http://localhost:8080/add");
+            request.addHeader("Content-Type", "application/json");
+            String pathFile = "src/test/resources/add-book-request.json";
+            String requestEntity = new String(Files.readAllBytes(Paths.get(pathFile)));
```

Comment:
You read this file multiple time across the test. Disk operations are slow so we need to reduce them as much as possible. You can move this to the test init logic to read it once.
Please note, that this comment is applicable to a bunch of other tests